observant-otters/forumthing/forum/urls.py

```diff
@@ -1,6 +1,20 @@
-from django.urls import path
+from django.urls import path, re_path
 from . import views
 
 urlpatterns = [
-    path('', views.home, name='forum-homepage'),
+    path(
+        "",
+        views.home,
+        name="forum-homepage"
+    ),
+    path(
+        'forum/threads',
+        views.threads,
+        name="threads-all"
+    ),
+    re_path(
+        r"^forum/threads/(?P<id>\d+)/$",
+        views.threads,
```

Comment:
To prevent nesting ifs too deep, it'd make sense to have this be its own view (that way, you don't have to check for the presence of the `id` parameter for each method).

Reply:
Also, I think `(?P<id>\d+)` can be simplified to `<int:id>`